index.js

```diff
@@ -0,0 +1,164 @@
+let taskArr = [];
+let start = 0;
+let current = 0;
+
+
+if (localStorage.getItem('localObject') !== null) {
+    taskArr = JSON.parse(localStorage.getItem('localObject'));
+} else {
+    taskArr = [];
+}
+
+function editListen() {
+const getEdit = document.querySelectorAll('p');
+getEdit.forEach((e, i) => {
+    e.addEventListener('keypress', (j) => {
+    if (j.key === 'Enter') {
+        const x = e.innerText;
+        edit(i, x);
+        show();
+        drag();
+        editListen();
+        delOne();
+    }
+    });
+});
+}
+
+function drag() {
+const getlistid = document.querySelectorAll('li');
+getlistid.forEach((e, i) => {
+    e.addEventListener('dragstart', () => {
+    start = i;
+    });
+    e.addEventListener('dragenter', () => {
+    current = i;
+    e.classList.add('bg-success');
+    });
+    e.addEventListener('dragleave', () => {
+    e.classList.remove('bg-success');
+    });
+    e.addEventListener('dragend', () => {
+    e.classList.remove('bg-primary');
+    if (start !== current) {
+        taskArr[start].index = current;
+        taskArr[current].index = start;
+        show();
+        drag();
+        editListen();
+        delOne();
+    }
+    });
+});
+}
+
+function clear() {
+const button = document.querySelector('.btn');
+button.addEventListener('click', () => {
+    const check = document.querySelectorAll('#check');
+    check.forEach((i) => {
+    if (i.checked) {
+        taskArr.splice(i, 1);
+        taskArr.forEach((x, y) => {
+        x.index = y;
```

Comment:
I would suggest creating a function to delete and update the index following the DRY principle.